tests/frontend_selenium/pages/farm.py

```diff
@@ -0,0 +1,271 @@
+        sorted.append((index1))
+      return name,sorted
+
+    def farm_table_sorting_by_pp_id(self):
+      WebDriverWait(self.browser, 30).until(EC.visibility_of_element_located(self.table_farm_name))      
+      id=[]
+      table = self.browser.find_elements(*self.table)
+      for i in range(1,len(table)+1):                                                                                       
+        element=(self.browser.find_element(By.XPATH, self.table_row + str(i) + ']/td[6]').text)
+        id.append(int(element))
+      id.sort()
+      self.browser.find_element(*self.pricing_policy_arrow).click()
+      sorted=[]
+      for i in range (1,len(table)+1):
+        index=(self.browser.find_element(By.XPATH, self.table_row + str(i) + ']/td[6]').text)
+        sorted.append(int(index))
+      return id,sorted,table
+
+    def farm_table_sorting_by_pp_id_up(self,id,table):  
+      id.reverse()
+      sorted=[]
+      self.browser.find_element(*self.pricing_policy_arrow).click()
+      for i in range (1,len(table)+1):
+        index1=(self.browser.find_element(By.XPATH, self.table_row + str(i) + ']/td[6]').text)
+        sorted.append(int(index1))
+      return id, sorted
+
+    def setup_farmpayout_address(self):
+      self.browser.find_element(*self.details_arrow).click()
+      self.browser.find_element(*self.add_v2_button).click()
+
+    def add_farmpayout_address(self, data):
+      self.browser.find_element(*self.path).send_keys(Keys.CONTROL + "a")
+      self.browser.find_element(*self.path).send_keys(Keys.DELETE)
+      self.browser.find_element(*self.path).send_keys(data)
+      return self.browser.find_element(*self.submit_button)
+    
+    def setup_gateway(self, data):
+      self.browser.find_element(*self.details_arrow).click()
+      self.browser.find_element(*self.public_ip_list).click()
+      self.browser.find_element(*self.add_ip_button).click()
+      self.browser.find_element(*self.gateway_text_field).send_keys(data)
+
+    def add_ip(self, data):
+      self.browser.find_element(*self.ip_text_field).send_keys(Keys.CONTROL + "a")
+      self.browser.find_element(*self.ip_text_field).send_keys(Keys.DELETE)
+      self.browser.find_element(*self.ip_text_field).send_keys(data) 
+      return self.browser.find_element(*self.save_button)
+
+    def setup_ip(self, data):
+      self.browser.find_element(*self.details_arrow).click()
+      self.browser.find_element(*self.public_ip_list).click()
+      self.browser.find_element(*self.add_ip_button).click()
+      self.browser.find_element(*self.ip_text_field).send_keys(data)
+
+    def add_gateway(self, data):
+      self.browser.find_element(*self.gateway_text_field).send_keys(Keys.CONTROL + "a")
+      self.browser.find_element(*self.gateway_text_field).send_keys(Keys.DELETE)
+      self.browser.find_element(*self.gateway_text_field).send_keys(data) 
+      return self.browser.find_element(*self.save_button)
+
+    def delete_ip(self):
```

Comment:
which ip will be deleted? nothing is sent to the method, is it random?

Reply:
in page `test_farm.py`, the function `test_delete_ip` create a farm and expand it then add an IP.
then this function use the same created farm and delete the IP that was added in it (so that each test is independent from each other).

Reply:
Changed the add and delete of ip, gateway and address to specify farm name to be modified.